src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.html

```diff
@@ -102,6 +103,16 @@
                     <label for="isPublic{{ key.id }}" class="cbx-label"></label>
                     <span *ngIf="key.status !== 'finished'"> N/A </span>
                   </td>
+
+                  <td>
+                    <a [matMenuTriggerFor]="settingsmenu"><i class="fa fa-gear"></i></a>
+                    <mat-menu #settingsmenu="matMenu">
+                        <button (click)="confirmSubmissionVisibility(key, key.is_public)" mat-menu-item>
+                          <mat-icon class="green-text">{{key.submissionVisibilityIcon}}</mat-icon>
+                          <span>{{key.submissionVisibilityText}}</span>
+                        </button>
```

Comment:
Indentation is little off here. See the `mat-menu` tag.

Reply:
done